### PR TITLE
ECMS-5417 XSS - executing script when edit a web content containing a li...

### PR DESCRIPTION
**Pull request by vudangngoc**
...nk with hexa characters

### CODE REVIEW

File: core/webui/src/main/java/org/exoplatform/ecm/webui/form/UIDialogForm.java [hunk 1]
```diff
@@ -1496,6 +1497,11 @@ public void addWYSIWYGField(String name, String[] arguments) throws Exception {
   }
 
   public void addRichtextField(String name, String label, String[] arguments) throws Exception {
+    if(arguments[0].contains("/node/default.html/jcr:content/jcr:data")){
+      String temp = arguments[arguments.length - 1];
```
Comment (anhvurz90) [inline core/webui/src/main/java/org/exoplatform/ecm/webui/form/UIDialogForm.java]: IMO escaping value here is quite risk and may raise side effect. We should escape only link title in the portal link plugin (js code)

File: core/webui/src/main/java/org/exoplatform/ecm/webui/utils/DialogFormUtil.java [hunk 2]
```diff
@@ -215,12 +216,15 @@
...
           }
         }
-      }
+      } 
     }
     Iterator iter = properties.values().iterator() ;
     JcrInputProperty property ;
     while (iter.hasNext()) {
       property = (JcrInputProperty) iter.next() ;
+      if("/node/default.html/jcr:content/jcr:data".equals(property.getJcrPath())){
```
Comment (anhvurz90) [inline core/webui/src/main/java/org/exoplatform/ecm/webui/utils/DialogFormUtil.java]: IMO escaping value here is quite risk and may raise side effect. We should escape only link title in the portal link plugin (js code)

File: core/webui/src/main/java/org/exoplatform/ecm/webui/form/UIDialogForm.java [hunk 3]
```diff
@@ -44,6 +44,7 @@
 import javax.jcr.ValueFormatException;
 import javax.jcr.lock.LockException;
 
+import org.apache.commons.lang.StringEscapeUtils;
```
Comment (trangvh) [inline core/webui/src/main/java/org/exoplatform/ecm/webui/form/UIDialogForm.java]: obsolete import

File: core/webui/src/main/java/org/exoplatform/ecm/webui/utils/DialogFormUtil.java [hunk 4]
```diff
@@ -31,6 +31,7 @@
 import javax.jcr.Property;
 import javax.jcr.PropertyType;
 
+import org.apache.commons.lang.StringEscapeUtils;
```
Comment (trangvh) [inline core/webui/src/main/java/org/exoplatform/ecm/webui/utils/DialogFormUtil.java]: obsolete import

File: core/webui/src/main/java/org/exoplatform/ecm/webui/utils/DialogFormUtil.java [hunk 5]
```diff
@@ -215,7 +216,7 @@
             }
           }
         }
-      }
+      }
```
Comment (trangvh) [inline core/webui/src/main/java/org/exoplatform/ecm/webui/utils/DialogFormUtil.java]: useless change.

**Pull request closed without merging**